tests/datatypes/test_str.py

```diff
@@ -695,6 +695,20 @@ def test_isidentifier(self):
             print(str_, str_.isidentifier())
             """)
 
+    def test_repr(self):
+        self.assertCodeExecution("""
+        str_ = "\\r\\n"
+        print(repr(str_))
+        """)
+
+    def test_splitlines(self):
+        self.assertCodeExecution("""
+        str_ = "aaa\\nbbb\\rccc\\r\\nddd\\n\\reee"
+
+        print(str_.splitlines())
+        print(str_.splitlines(True))
```

Comment:
Hey @seocam, thanks for the PR!

One little thing: the empty string case isn't handled correctly, it's returning a list containing an empty string like this:

```
>>> ''.splitlines()
['']
```

It should return an empty list, like this:

```
$ python
Python 3.4.4 (default, Nov 13 2016, 11:58:00) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> ''.splitlines()
[]
```

Can you add the special handling and add this to the tests too?
Thank you!